src/os/bluestore/bluestore_tool.cc

```diff
@@ -72,6 +73,53 @@ void validate_path(CephContext *cct, const string& path, bool bluefs)
   }
 }
 
+struct BlueStorePair
```

Comment:
There is the `static_ptr<...>` template that does exactly this without the need for dynamic allocation.
On the other hand, the current approach is so _explicitly hackish_ I like it while nobody really cares dynallocs in the tools.

Reply:
What good does come from using `static_ptr` ?
The problem is that bluestore and bluestore-rdr are 2 completely independent classes.
Granted, I can keep it in one place; but then I would need to `reinterpret_cast` the location.

I agree that the current solution is _crude and primitive_ but not _hackish_.